packages/ra-core/src/dataProvider/useGetOne.ts

```diff
@@ -16,11 +16,16 @@ import useDataProvider from './useDataProvider';
  * with the same parameters, until the response arrives.
  *
  * @param resource The resource name, e.g. 'posts'
- * @param id The resource identifier, e.g. 123
- * @param {Object} options Options object to pass to the react-query queryClient.
- * @param {boolean} options.enabled Flag to conditionally run the query. If it's false, the query will not run
- * @param {Function} options.onSuccess Side effect function to be executed upon success, e.g. { onSuccess: { refresh: true } }
- * @param {Function} options.onError Side effect function to be executed upon failure, e.g. { onError: error => notify(error.message) }
+ * @param {Params} params The getOne parameters { id }, e.g. { id: 123 }
+ * @param {Options} options Options object to pass to the react-query queryClient.
+ *
+ * @typedef Params
+ * @property id a resource identifier, e.g. 123
+ *
+ * @typedef Options
+ * @property enabled Flag to conditionally run the query. If it's false, the query will not run
+ * @property onSuccess Side effect function to be executed upon success, e.g. { onSuccess: { refresh: true } }
+ * @property onError Side effect function to be executed upon failure, e.g. { onError: error => notify(error.message) }
```

Comment:
Why a typedef when we have TS?

Reply:
because everyione doesn't have TS, and I think the jsDoc renders *before* type definitions in IDEs.